executor/point_get.go

```diff
@@ -383,28 +390,68 @@ func (e *PointGetExecutor) getAndLock(ctx context.Context, key kv.Key) (val []by
 }
 
 func (e *PointGetExecutor) lockKeyIfNeeded(ctx context.Context, key []byte) error {
+	_, err := e.lockKeyBase(ctx, key, false)
+	return err
+}
+
+// lockKeyIfExists locks the key if needed, but won't lock the key if it doesn't exis.
+// Returns the value of the key if the key exist.
+func (e *PointGetExecutor) lockKeyIfExists(ctx context.Context, key []byte) ([]byte, error) {
+	return e.lockKeyBase(ctx, key, true)
+}
+
+func (e *PointGetExecutor) lockKeyBase(ctx context.Context,
+	key []byte,
+	LockOnlyIfExists bool) ([]byte, error) {
 	if len(key) == 0 {
-		return nil
+		return nil, nil
 	}
+
 	if e.lock {
 		seVars := e.ctx.GetSessionVars()
 		lockCtx, err := newLockCtx(e.ctx, e.lockWaitTime, 1)
 		if err != nil {
-			return err
+			return nil, err
 		}
+		lockCtx.LockOnlyIfExists = LockOnlyIfExists
 		lockCtx.InitReturnValues(1)
 		err = doLockKeys(ctx, e.ctx, lockCtx, key)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		lockCtx.IterateValuesNotLocked(func(k, v []byte) {
 			seVars.TxnCtx.SetPessimisticLockCache(k, v)
 		})
 		if len(e.handleVal) > 0 {
 			seVars.TxnCtx.SetPessimisticLockCache(e.idxKey, e.handleVal)
 		}
+		if LockOnlyIfExists {
+			return e.getValueFromLockCtx(ctx, lockCtx, key)
+		}
 	}
-	return nil
+
+	return nil, nil
+}
+
+func (e *PointGetExecutor) getValueFromLockCtx(ctx context.Context,
```

Comment:
I think it might be better to put the function of extracting values from LockCtx in client-go. But this is acceptable.

Reply:
+1